src/platform/osx.rs

```diff
@@ -269,24 +267,30 @@ impl<'clipboard> Set<'clipboard> {
 		let pixels = data.bytes.into();
 		let image = image_from_pixels(pixels, data.width, data.height)
 			.map_err(|_| Error::ConversionFailure)?;
+
+		self.clipboard.clear();
+
 		let objects: Id<NSArray<NSObject, Owned>> = NSArray::from_vec(vec![image]);
-		let _: usize = unsafe { msg_send![self.pasteboard, clearContents] };
-		let success: BOOL = unsafe { msg_send![self.pasteboard, writeObjects: objects] };
-		if success == NO {
-			return Err(Error::Unknown {
+		let success: bool = unsafe { msg_send![self.clipboard.pasteboard, writeObjects: objects] };
+		if success {
+			Ok(())
+		} else {
+			Err(Error::Unknown {
 				description:
 					"Failed to write the image to the pasteboard (`writeObjects` returned NO)."
 						.into(),
-			});
+			})
 		}
-		Ok(())
 	}
 }
 
-// this is a convenience function that both cocoa-rs and
-//  glutin define, which seems to depend on the fact that
-//  Option::None has the same representation as a null pointer
-#[inline]
-fn class(name: &str) -> *mut Class {
-	unsafe { transmute(Class::get(name)) }
+/// Convenience function to get an Objective-C object from a
+/// specific class.
+fn object_class(class: &'static Class) -> Id<NSObject> {
+	// SAFETY: `Class` is valid to transmute into a pointer and the pointer
+	// is a valid object, so its ID can be obtained.
+	let class_id: Id<Class> = unsafe { Id::from_ptr(std::mem::transmute(class)) };
+
+	// SAFETY: All of the variants constructed are valid NSObjects.
+	unsafe { std::mem::transmute(class_id) }
 }
```

Comment:
Why the `transmute` instead of pointer casts?

```suggestion
fn object_class(class: &'static Class) -> Id<NSObject> {
	// SAFETY: `Class` is a valid object and `Id` will not mutate it
	unsafe { Id::from_ptr(class as *const _ as *mut _) }
}
```

Reply:
I believe you're 💯 correct here. The transmutes made more sense in a previous revision of this code which utilized less-typed code but with the current implementation, a pointer cast is much clearer.

Reply:
Changed, thanks again for the suggestion.